### ISSUE TITLE
Ctrl + Z do closeActiveEditor instead of Undo

### ISSUE THREAD

**Issue by Toug19** [microsoft/vscode#198919]
Type: <b>Bug</b>

Hi,
Sometimes, when I do Ctrl + z, it closes the active éditor instead of doing "Undo".
Seems that Vscode detect correct keys as I activated the "Developer: Toggle Keyboard Shortcuts Troubleshooting" and here are the logs:

2023-11-23 09:22:17.387 [info] [KeybindingService]: / Soft dispatching keyboard event
2023-11-23 09:22:17.387 [info] [KeybindingService]: \ Keyboard event cannot be dispatched
2023-11-23 09:22:17.388 [info] [KeybindingService]: / Received  keydown event - modifiers: [ctrl], code: ControlLeft, keyCode: 17, key: Control
2023-11-23 09:22:17.388 [info] [KeybindingService]: | Converted keydown event - modifiers: [ctrl], code: ControlLeft, keyCode: 5 ('Ctrl')
2023-11-23 09:22:17.389 [info] [KeybindingService]: \ Keyboard event cannot be dispatched in keydown phase.
2023-11-23 09:22:17.583 [info] [KeybindingService]: / Soft dispatching keyboard event
2023-11-23 09:22:17.583 [info] [KeybindingService]: | Resolving ctrl+[KeyW]
2023-11-23 09:22:17.583 [info] [KeybindingService]: \ From 3 keybinding entries, matched workbench.action.closeActiveEditor, when: no when condition, source: built-in.
2023-11-23 09:22:17.584 [info] [KeybindingService]: / Received  keydown event - modifiers: [ctrl], code: KeyW, keyCode: 90, key: z
2023-11-23 09:22:17.584 [info] [KeybindingService]: | Converted keydown event - modifiers: [ctrl], code: KeyW, keyCode: 56 ('Z')
2023-11-23 09:22:17.584 [info] [KeybindingService]: | Resolving ctrl+[KeyW]
2023-11-23 09:22:17.584 [info] [KeybindingService]: \ From 3 keybinding entries, matched workbench.action.closeActiveEditor, when: no when condition, source: built-in.
2023-11-23 09:22:17.584 [info] [KeybindingService]: + Invoking command workbench.action.closeActiveEditor.
2023-11-23 09:22:17.836 [info] [KeybindingService]: + Ignoring single modifier ctrl due to it being pressed together with other keys.

I checked the shortcut mapping rules on Preferences -> Keyboard Shortcuts and Undo is mapped to Ctrl + Z and closing editor to Ctrl+W.
Here are the copy paste of the settings:

{
  "key": "ctrl+w",
  "command": "workbench.action.closeActiveEditor"
}
{
  "key": "ctrl+z",
  "command": "undo"
}

Seems in the logs that both CTRL + W and CTRL + Z are detected...
I use a french keyboard with french layout (Z and W are  reversed from US keyboard)

Regards,
Sébastien.

VS Code version: Code 1.84.2 (1a5daa3a0231a0fbba4f14db7ec463cf99d7768e, 2023-11-09T10:50:47.800Z)
OS version: Linux x64 5.4.0-153-generic
Modes:
Remote OS version: Linux x64 5.4.0-153-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i7-11850H @ 2.50GHz (6 x 2496)|
|GPU Status|2d_canvas: unavailable_software<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: disabled_software<br>multiple_raster_threads: enabled_on<br>opengl: disabled_off<br>rasterization: disabled_software<br>raw_draw: disabled_off_ok<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: unavailable_software<br>webgl2: unavailable_software<br>webgpu: disabled_off|
|Load (avg)|1, 0, 0|
|Memory (System)|15.61GB (12.44GB free)|
|Process Argv|--unity-launch --crash-reporter-id 17e239a0-2837-42dd-ac4d-c2e41eb2ce26|
|Screen Reader|no|
|VM|57%|
|DESKTOP_SESSION|xfce|
|XDG_CURRENT_DESKTOP|XFCE|
|XDG_SESSION_DESKTOP|xfce|
|XDG_SESSION_TYPE|x11|

|Item|Value|
|---|---|
|Remote|Dev Container: VC HPC devcontainer|
|OS|Linux x64 5.4.0-153-generic|
|CPUs|11th Gen Intel(R) Core(TM) i7-11850H @ 2.50GHz (6 x 2496)|
|Memory (System)|15.61GB (12.44GB free)|
|VM|0%|
</details><details><summary>Extensions (12)</summary>

Extension|Author (truncated)|Version
---|---|---
remote-containers|ms-|0.321.0
markdown-mermaid|bie|1.20.0
bitbake|Eug|1.1.2
gtest-snippets|idm|1.0.1
python|ms-|2023.20.0
vscode-pylance|ms-|2023.11.10
cmake-tools|ms-|1.16.31
cpptools|ms-|1.18.5
cpptools-extension-pack|ms-|1.3.0
jinjahtml|sam|0.20.0
code-spell-checker|str|3.0.1
cmake|twx|0.0.17

(2 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
py29gd2263cf:30880073
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30695312
azure-dev_surveyone:30548225
3biah626:30602489
89544117:30613380
showlangstatbar:30737416
pythonfmttext:30731395
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
welcomedialog:30887143
pythonnosmt12:30797651
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30869155
dsvsc013:30795093
dsvsc014:30804076
dsvsc015:30845448
pythontestfixtcf:30871695
pythonregdiag2:30871582
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
pythontbext0:30879054
dsvsc016:30886110
dsvsc017cf:30886113
dsvsc018:30886114
aa_t_chat:30882232

```

</details>

**Comment by ulugbekna** [comment]
Based on the description you provided, this works as expected? 

Please, have a look at our [reference](https://github.com/microsoft/vscode/wiki/Keybinding-Issues) to troubleshoot keyboard issues.

**Comment by Toug19** [comment]
It doesn't! It closes the editor wheras it should only make undo!

**Comment by ulugbekna** [comment]
Have you read the reference I linked above?

**Comment by Toug19** [comment]
The information you request are on the first message.

**Comment by ulugbekna** [comment]
> Sometimes, when I do Ctrl + z, it closes the active éditor instead of doing "Undo".

VS Code cannot "sometimes" differently dispatch keybindings.

---

> Sometimes, when I do Ctrl + z, it closes the active éditor instead of doing "Undo".
Seems that Vscode detect correct keys as I activated the "Developer: Toggle Keyboard Shortcuts Troubleshooting" and here are the logs:

You describe pressing `ctrl+z`, then say "detect correct keys", then attach logs that show "ctrl+w" being detected as pressed. This is confusing. 

---

The first entry in the reference I linked will likely fix this problem

> Are you connecting to the Linux machine via some form of virtualization or remote desktop software?
> symptoms: e.g. pressing AltGr acts as Backspace, pressing C acts as ArrowLeft, etc.
solution: use "keyboard.dispatch": "keyCode" in your settings and restart VS Code.
explanation: VS Code on Linux dispatches keybindings using scan codes to accomodate various keyboard layouts. This works well when sitting physically at the machine and the OS is the one creating keyboard events, but some virtualization / remote desktop software creates keyboard events with incorrect scan codes.
